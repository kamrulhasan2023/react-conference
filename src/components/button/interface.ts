import React from 'react';
import { IconNames } from './../icons/interface';

export const buttonSizes = ['lg', 'md', 'sm', 'xs'] as const;
type Size = (typeof buttonSizes)[number];

export interface IFillButton {
  label: string;
  colorScheme: ColorScheme;
  size: Size;
  clickHandler: (
    e: React.MouseEvent<HTMLElement> | React.FormEvent<HTMLFormElement>
  ) => void;
  hoverColorScheme?: ColorScheme;
  iconName?: IconNames;
  iconIsLeft?: boolean;
  isDisabled?: boolean;
  variant?: 'fill';
  type?: 'button';
}

export interface IFillSubmitButton
  extends Omit<IFillButton, 'clickHandler' | 'type'> {
  type: 'submit';
}

export interface IFillRedirectButton
  extends Omit<IFillButton, 'clickHandler' | 'isDisabled'> {
  href?: string;
  isOpenNewTab: boolean;
}

export interface IOutlineButton
  extends Omit<IFillButton, 'colorScheme' | 'hoverColorScheme' | 'variant'> {
  colorScheme: OutlineColorScheme;
  hoverColorScheme: OutlineColorScheme;
  variant: 'outline';
}

export interface IOutlineSubmitButton
  extends Omit<IOutlineButton, 'clickHandler' | 'type'> {
  type: 'submit';
}

export interface IOutlineRedirectButton
  extends Omit<IOutlineButton, 'clickHandler' | 'isDisabled'> {
  href?: string;
  isOpenNewTab: boolean;
}

export type IBaseButton =
  | Omit<IOutlineButton, 'clickHandler' | 'type'>
  | Omit<IFillButton, 'clickHandler' | 'type'>;

export const buttonColorSchemes = [
  'bg-primary-text-white',
  'bg-success-text-white',
  'bg-black-text-white',
  'bg-white-text-black',
] as const;
type ColorScheme = (typeof buttonColorSchemes)[number];

export const outlineButtonColorSchemes = [
  'bg-transparent-text-primary',
  'bg-transparent-text-success',
] as const;
type OutlineColorScheme = (typeof outlineButtonColorSchemes)[number];
