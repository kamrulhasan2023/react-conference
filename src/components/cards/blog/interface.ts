import { IDimension } from 'src/components/custom-image/interface';

export interface IBlogCard {
  href: string;
  image: {
    src: string;
    alt: string;
    lqip: string;
  };
  title: string;
  description: string;
  tags: {
    bgColor: { hex: string };
    fontColor: { hex: string };
    title: string;
  }[];
  isFeatured: boolean;
}

export const BLOG_IMAGE_DIMENSION: IDimension = {
  width: 960,
  height: 720,
};
