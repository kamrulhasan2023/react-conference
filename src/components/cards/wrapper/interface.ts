export interface ICardWrapper {
  cardShadow: ICardShadow;
  hoverCardShadow?: ICardShadow;
  cardBorderRadius: ICardBorderRadius;
}

export const cardWrapperShadow = ['shadow-one', 'shadow-none'] as const;
type ICardShadow = (typeof cardWrapperShadow)[number];

export const cardWrapperBorderRadius = [
  'radius-one',
  'radius-two',
  'radius-none',
] as const;
type ICardBorderRadius = (typeof cardWrapperBorderRadius)[number];
