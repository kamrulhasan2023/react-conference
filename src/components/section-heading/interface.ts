import { ISanityCustomPortableText } from 'src/backend/queries/portable-text/interface';

export const sectionHeadingColorschemes = [
  'primary',
  'success',
  'white',
  'black',
] as const;

type ColorScheme = (typeof sectionHeadingColorschemes)[number];
export interface ISectionHeading {
  title: string;
  colorScheme: ColorScheme;
  isCenter: boolean;
  hasBottomSpacing: boolean;
  subtitle?: string;
  description?: ISanityCustomPortableText;
}
