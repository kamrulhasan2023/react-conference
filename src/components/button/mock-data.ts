import {
  IFillButton,
  IFillRedirectButton,
  IOutlineButton,
  IOutlineRedirectButton,
} from './interface';

export const buttonProps: IFillButton = {
  label: 'Button',
  colorScheme: 'bg-primary-text-white',
  hoverColorScheme: 'bg-success-text-white',
  size: 'lg',
  clickHandler: () => {},
  variant: 'fill',
};

export const redirectButtonProps: IFillRedirectButton = {
  href: 'https://www.google.com/',
  isOpenNewTab: true,
  label: 'Redirect Button',
  colorScheme: 'bg-primary-text-white',
  hoverColorScheme: 'bg-success-text-white',
  size: 'lg',
  variant: 'fill',
};

export const outlineButtonProps: IOutlineButton = {
  label: 'Outline Button',
  colorScheme: 'bg-transparent-text-primary',
  hoverColorScheme: 'bg-transparent-text-success',
  size: 'lg',
  clickHandler: () => {},
  variant: 'outline',
};

export const redirectOutlineButtonProps: IOutlineRedirectButton = {
  href: 'https://www.google.com/',
  isOpenNewTab: true,
  label: 'Outline Redirect Button',
  colorScheme: 'bg-transparent-text-primary',
  hoverColorScheme: 'bg-transparent-text-success',
  size: 'lg',
  variant: 'outline',
};
