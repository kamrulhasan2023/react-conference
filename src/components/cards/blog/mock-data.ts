import { IBlogCard } from './interface';

export const blogCardProps: IBlogCard = {
  href: 'https://www.lemonhive.com/',
  image: {
    src: 'https://res.cloudinary.com/dktw59lc2/image/upload/v1669267967/BrightonSEO/Carousel/image-2_yph8q6.jpg',
    alt: 'dummy image 2',
    lqip: 'https://res.cloudinary.com/dktw59lc2/image/upload/v1669267967/BrightonSEO/Carousel/image-2_yph8q6.jpg',
  },
  title: 'Headless Website Development',
  description:
    'Headless solutions built with Sanity.io, Strapi, Prismic, Contentful and others with front-end frameworks such as Next.js, Gatsby, & Nuxt.',
  tags: [
    { fontColor: { hex: 'black' }, bgColor: { hex: 'red' }, title: 'Reactjs' },
    { fontColor: { hex: 'black' }, bgColor: { hex: 'red' }, title: 'Nextjs' },
  ],
  isFeatured: false,
};
