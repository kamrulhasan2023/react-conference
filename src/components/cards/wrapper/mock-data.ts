import { ICardWrapper } from './interface';

export const cardWrapperProps: ICardWrapper = {
  cardBorderRadius: 'radius-none',
  cardShadow: 'shadow-one',
};
