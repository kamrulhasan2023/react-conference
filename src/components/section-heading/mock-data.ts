import { ISectionHeading } from './interface';

export const sectionDescriptionProps = [
  {
    _key: '0ffdc2e0f0b8',
    _type: 'block',
    children: [
      {
        _key: '9ba941b1390f',
        _type: 'span',
        marks: ['805b62da0516'],
        text: 'Headless website',
      },
      {
        _key: '9a6ae5c5c320',
        _type: 'span',
        marks: [],
        text: ' for an award winning digital marketing agency.',
      },
    ],
    markDefs: [
      {
        _key: '805b62da0516',
        _type: 'link',
        href: 'https://www.google.com',
        linkType: 'externalLink',
      },
    ],
    style: 'normal',
  },
  {
    _key: '4ed7297d9dcc',
    _type: 'block',
    children: [
      {
        _key: 'da46c7671dee',
        _type: 'span',
        marks: [],
        text: '',
      },
    ],
    markDefs: [],
    style: 'normal',
  },
  {
    _key: 'c0b3d76ed804',
    _type: 'block',
    children: [
      {
        _key: '1d4ab47cbb880',
        _type: 'span',
        marks: ['strong'],
        text: 'Lorem ipsum',
      },
      {
        _key: '1b8f25fb6ad2',
        _type: 'span',
        marks: [],
        text: ' dolor ',
      },
      {
        _key: '69ae9b1fb5c7',
        _type: 'span',
        marks: ['em'],
        text: 'sit amet consectetur',
      },
      {
        _key: '2e48bd8c68e6',
        _type: 'span',
        marks: [],
        text: '. Vitae cursus ultricies interdu ridiculus. Nullam nuwel craegestas risus ',
      },
      {
        _key: 'c0b66fa9a5e6',
        _type: 'span',
        marks: ['underline'],
        text: 'diam lorem nascetur',
      },
      {
        _key: 'c751dbb05102',
        _type: 'span',
        marks: [],
        text: '. Lorem ipsum dolor sit amet consectetur. Vitae cursus ultricies interdu ridiculus. Nullam nuwel craegestas risus ',
      },
      {
        _key: '95e7c8d27766',
        _type: 'span',
        marks: ['strong'],
        text: 'diam lorem nascetu ullam nuwel',
      },
    ],
    markDefs: [],
    style: 'normal',
  },
];

export const sectionHeadingProps: ISectionHeading = {
  subtitle: 'Lemon Hive',
  title: 'What Makes Us Different?',
  description: sectionDescriptionProps,
  colorScheme: 'primary',
  hasBottomSpacing: true,
  isCenter: false,
};
