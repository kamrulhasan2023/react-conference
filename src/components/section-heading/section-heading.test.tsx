import { render } from '@testing-library/react';
import SectionHeading from '.';
import { sectionHeadingProps } from './mock-data';
import { sectionHeadingColorschemes } from './interface';

it('renders section heading correctly', () => {
  const { getByRole } = render(<SectionHeading {...sectionHeadingProps} />);
  const element = getByRole('presentation');
  expect(element).toBeInTheDocument();
  expect(element).toMatchSnapshot();
});

describe.each(sectionHeadingColorschemes)(
  'Given section heading colorScheme: %s',
  (colorScheme) => {
    it(`Expect section heading to have class ${colorScheme}`, () => {
      const { getByRole } = render(
        <SectionHeading {...sectionHeadingProps} colorScheme={colorScheme} />
      );
      const element = getByRole('presentation');
      expect(element).toHaveClass(colorScheme);
    });
  }
);

describe.each([true, false])(
  'Given section heading alignment: %s',
  (isCenter) => {
    it(`Expect section heading to have class ${isCenter}`, () => {
      const { getByRole } = render(
        <SectionHeading {...sectionHeadingProps} isCenter={isCenter} />
      );
      const element = getByRole('presentation');
      if (isCenter) {
        expect(element).toHaveClass('text-center');
      } else {
        expect(element).not.toHaveClass('text-center');
      }
    });
  }
);

const bottomSpacingClasses = `mb-8 md:mb-12`;

describe.each([true, false])(
  'Given section heading bottom spacing: %s',
  (hasBottomSpacing) => {
    it(`Expect section heading to have class ${hasBottomSpacing}`, () => {
      const { getByRole } = render(
        <SectionHeading
          {...sectionHeadingProps}
          hasBottomSpacing={hasBottomSpacing}
        />
      );
      const element = getByRole('presentation');
      if (hasBottomSpacing) {
        expect(element).toHaveClass(bottomSpacingClasses);
      } else {
        expect(element).not.toHaveClass(bottomSpacingClasses);
      }
    });
  }
);
