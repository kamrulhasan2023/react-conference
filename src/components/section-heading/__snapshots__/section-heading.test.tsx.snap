// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders section heading correctly 1`] = `
<div
  class="mb-8 md:mb-12 section-heading primary"
  role="presentation"
>
  <span
    class="mb-2 block text-lg font-medium md:text-xl sub-title"
  >
    Lemon Hive
  </span>
  <h2
    class="text-2xl font-medium  md:text-4xl md:font-bold title"
  >
    What Makes Us Different?
  </h2>
  <div
    class="mt-4 lg:mt-8"
  >
    <article
      class="normal-block-font-size-normal"
    >
      <div
        class="space-y-4 main-wrapper"
      >
        <p
          class="normal"
          id="0ffdc2e0f0b8"
        >
          <a
            class="underline text-secondary"
            href="https://www.google.com"
            target="_blank"
          >
            Headless website
          </a>
           for an award winning digital marketing agency.
        </p>
        <br />
        <p
          class="normal"
          id="c0b3d76ed804"
        >
          <strong>
            Lorem ipsum
          </strong>
           dolor 
          <em>
            sit amet consectetur
          </em>
          . Vitae cursus ultricies interdu ridiculus. Nullam nuwel craegestas risus 
          <span
            style="text-decoration: underline;"
          >
            diam lorem nascetur
          </span>
          . Lorem ipsum dolor sit amet consectetur. Vitae cursus ultricies interdu ridiculus. Nullam nuwel craegestas risus 
          <strong>
            diam lorem nascetu ullam nuwel
          </strong>
        </p>
      </div>
    </article>
  </div>
</div>
`;
